# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: js/src/jit/Ion.cpp
# Commit: f0fc28562ec4
# Full Hash: f0fc28562ec4396e455f723837906d60b03f4311
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2021-12-15 04:41:24
# Description:
#   Bug 1745388: Compute dependency info for shape guard removal r=jandem
#   
#   If LICM and GVN are both disabled, `ins->dependency()` is never set and we dereference a null pointer.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D133562
# ==============================================================================

diff -r fa1e33c47711 -r f0fc28562ec4 js/src/jit/Ion.cpp
--- a/js/src/jit/Ion.cpp	Tue Dec 14 17:04:19 2021 +0000
+++ b/js/src/jit/Ion.cpp	Tue Dec 14 17:17:26 2021 +0000
@@ -1143,9 +1143,11 @@
   ValueNumberer gvn(mir, graph);
 
   // Alias analysis is required for LICM and GVN so that we don't move
-  // loads across stores.
+  // loads across stores. We also use alias information when removing
+  // redundant shapeguards.
   if (mir->optimizationInfo().licmEnabled() ||
-      mir->optimizationInfo().gvnEnabled()) {
+      mir->optimizationInfo().gvnEnabled() ||
+      mir->optimizationInfo().eliminateRedundantShapeGuardsEnabled()) {
     {
       AutoTraceLog log(logger, TraceLogger_AliasAnalysis);
 
